ci.sh

```diff
@@ -24,4 +24,15 @@ DIR_THIS="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 export TARGET_REPO_PATH=${TARGET_REPO_PATH:-$(pwd)}
 export TARGET_REPO_NAME=${TARGET_REPO_NAME:-${TARGET_REPO_PATH##*/}}
 
+function upload_coverage_report {
```

Comment:
Why does ci.sh use a different upload function than gitlab.sh?

Reply:
The main reason is because of the VM and docker used by the CI.
- **Github Action** it's better to use the [Codecov Github Action](https://github.com/codecov/codecov-action) because direct upload using shell report will always be on master branch. Github Action allows installing python3 directly.
- **Gitlab CI** if use the old docker-in-docker will be an alphine image, thus the python set up will be a bit different documented as follows. Furthermore, installation can only happen in before_script
>     * Coveralls
>       ```yaml
>       before_script:
>         - apk add --update bash coreutils tar
>         - apk add --update python3 py3-pip python3-dev
>         - git clone --quiet --depth 1 https://github.com/ros-industrial/industrial_ci .industrial_ci -b master
>       ...
>       variables:
>         CODE_COVERGAE: "coveralls.io"
>         COVERALLS_REPO_TOKEN: "..."
>       ```
- **Travis CI:** default in xenial only python2 installed, and installing python3 needs to be inside **before_install**.

There are 2 ways I can go around this
1. Remove the pip dependency for coveralls, use only ruby command line tools.

   This will need to convert the python report format `.coverage` or `coverage.xml` into `LCOV` inside the docker, which additional external module need to written. (This is actually inside my TODO list)

1. Check which python version can be installed

   This might be the quicker solution for now. Maybe I should go with this one first?